tests/letstest/scripts/test_leauto_upgrades.sh

```diff
@@ -15,19 +15,56 @@ if ! command -v git ; then
         exit 1
     fi
 fi
-BRANCH=`git rev-parse --abbrev-ref HEAD`
 # 0.5.0 is the oldest version of letsencrypt-auto that can be used because it's
 # the first version that pins package versions, properly supports
 # --no-self-upgrade, and works with newer versions of pip.
-git checkout -f v0.5.0
+git checkout -f v0.5.0 letsencrypt-auto
 if ! ./letsencrypt-auto -v --debug --version --no-self-upgrade 2>&1 | grep 0.5.0 ; then
     echo initial installation appeared to fail
     exit 1
 fi
 
-git checkout -f "$BRANCH"
-EXPECTED_VERSION=$(grep -m1 LE_AUTO_VERSION letsencrypt-auto | cut -d\" -f2)
-if ! ./letsencrypt-auto -v --debug --version --no-self-upgrade 2>&1 | grep $EXPECTED_VERSION ; then
+# Now that python and openssl have been installed, we can set up a fake server
+# to provide a new version of letsencrypt-auto. First, we start the server and
+# directory to be served.
+MY_TEMP_DIR=$(mktemp -d)
+PORT_FILE="$MY_TEMP_DIR/port"
+SERVER_PATH=$(tools/readlink.py tools/simple_http_server.py)
+cd "$MY_TEMP_DIR"
+"$SERVER_PATH" 0 > $PORT_FILE &
```

Comment:
`python "$SERVER_PATH"`?

Reply:
The file is executable with `/usr/bin/env python` at the top so these should be equivalent.

Reply:
N I C E.